src/main/java/uk/gov/justice/hmpps/prison/service/OffenderAssessmentService.java

```diff
@@ -120,20 +124,39 @@ private AssessmentQuestion getAssessmentQuestionAndAnswers(final AssessmentEntry
             answers.stream().skip(1).collect(toList()));
     }
 
-    public List<AssessmentClassification> getOffendersAssessmentRatings(List<String> offenderList) {
-        return offenderList.stream().map(this::getOffenderCurrentAssessmentRating).collect(Collectors.toList());
+    @Transactional(readOnly = true)
+    public List<AssessmentClassification> getOffendersAssessmentRatings(final List<String> offenderNos) {
+        final var classifications = new ArrayList<AssessmentClassification>();
+        final var batch = Lists.partition(new ArrayList<>(offenderNos), maxBatchSize);
+        batch.forEach(offenderNosBatch -> classifications.addAll(getOffendersCurrentAssessmentRating(offenderNosBatch)));
```

Comment:
Meh, could be a flatMap to keep up with consistency

Reply:
Yes, that was the comment you had before and i fixed it... I'll do it again